File: Trunk/vpPriV100Filopa/InditexFilopa/Base/FichaCliente/BasIsFichaCliente.cs
using Generico;
using Primavera.Extensibility.Base.Editors;
using Primavera.Extensibility.BusinessEntities;
using Primavera.Extensibility.BusinessEntities.ExtensibilityService;
using Primavera.Extensibility.BusinessEntities.ExtensibilityService.EventArgs;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace InditexFilopa
{
    public class BasIsFichaCliente : FichaClientes
    {

        public override void TeclaPressionada(int KeyCode, int Shift, ExtensibilityEventArgs e)
        {
            base.TeclaPressionada(KeyCode, Shift, e);

            // Bruno Peixoto 27/04/2020 - CTRL+O para abrir o formumlario de Fiações Inditex
            if (KeyCode == 79 & this.Cliente.Inactivo == false)
            {
                Module1.certFiacoes = this.Cliente.Cliente;

                ExtensibilityResult result = BSO.Extensibility.CreateCustomFormInstance(typeof(FrmInditexFilopaView));
                FrmInditexFilopaView frm = result.Result;
                frm.ShowDialog();
            }

            if (KeyCode == 79 & this.Cliente.Inactivo == true)
                MessageBox.Show("Cliente Anulado! Não é possível abrir o formulário de Fiações Inditex!", "", MessageBoxButtons.OK, MessageBoxIcon.Exclamation);

        }

    }
}

File: Branches/VBInicial/vpPriV100GrupoMundifios/InditexFilopa/Base/FichaCliente/BasIsFichaCliente.vb
Imports Primavera.Extensibility.Base.Editors
Imports Primavera.Extensibility.BusinessEntities.ExtensibilityService.EventArgs
Imports StdBE100
Imports Vimaponto.PrimaveraV100.Clientes.GrupoMundifios.Generico

Namespace InditexFilopa
    Public Class BasIsFichaCliente
        Inherits FichaClientes

        'Public Overrides Sub TeclaPressionada(KeyCode As Integer, Shift As Integer, e As ExtensibilityEventArgs)
        '    MyBase.TeclaPressionada(KeyCode, Shift, e)

        '    'Bruno Peixoto 27/04/2020 - CTRL+O para abrir o formumlario de Fiações Inditex
        '    If KeyCode = 79 And Me.Cliente.Inactivo = False Then
        '        Module1.certFiacoes = Me.Cliente.Cliente

        '        Load FrmInditexCliente
        'FrmInditexCliente.Show
        '    End If

        '    If KeyCode = 79 And Me.Cliente.Inactivo = True Then
        '        MsgBox("Cliente Anulado! Não é possível abrir o formulário de Fiações Inditex!", vbInformation)
        '    End If

        'End Sub

    End Class
End Namespace